duct-tape/src/memory.c

```diff
@@ -594,8 +594,12 @@ kern_return_t mach_vm_map_external(vm_map_t target_map, mach_vm_offset_t* addres
 	dtape_stub_unsafe();
 };
 
-kern_return_t mach_vm_msync(vm_map_t map, mach_vm_address_t address, mach_vm_size_t size, vm_sync_t sync_flags) {
-	dtape_stub_unsafe();
+kern_return_t mach_vm_msync(vm_map_t map, mach_vm_offset_t address, mach_vm_size_t size, vm_sync_t sync_flags) {
+	if (!dtape_hooks->task_change_protection(map->dtape_task->context, address, size, sync_flags)) {
```

Comment:
Seems like a typo leftover from copying the mprotect implementation.

But also, we don't want to give the Linux call the Mach flags (they have different values). Let's translate them.

```suggestion
	int linux_flags = 0;

	// TODO: give the Linux bits names/macros

	if (sync_flags & VM_SYNC_ASYNCHRONOUS) {
		linux_flags |= 1 << 0;
	}

	if (sync_flags & VM_SYNC_SYNCHRONOUS) {
		linux_flags |= 1 << 2;
	}

	if (sync_flags & VM_SYNC_INVALIDATE) {
		linux_flags |= 1 << 1;
	}

	if (!dtape_hooks->task_sync_memory(map->dtape_task->context, address, size, linux_flags)) {
```